### DISCUSSION TITLE
onCacheEntryAdded works only the first time?

### DISCUSSION THREAD

**Discussion by mad-coder-365** [reduxjs/redux-toolkit#3017] [Q&A] [answered]
Hi am new to Streaming Updates, please help with this

I have a query to get the user's info
```
getUserInfo: builder.query<IUserInfo | null, any>({
            queryFn: () => ({data: null}),
            keepUnusedDataFor: 30,
            async onCacheEntryAdded(
                arg,
                {updateCachedData, cacheDataLoaded, cacheEntryRemoved}
            ) {
                const socket = getSocket()
                try {
                    await cacheDataLoaded

                    console.log('working...')

                    socket.emit('getUserInfo', null, (info) => {
                        console.log(info)
                        updateCachedData(() => {
                            return info
                        })
                    })
                } catch {
                    // no-op in case `cacheEntryRemoved` resolves before `cacheDataLoaded`,
                    // in which case `cacheDataLoaded` will throw
                }
                await cacheEntryRemoved
            },
        }),
```
then in a component, I am using it like
```
const {data: info} = useGetUserInfoQuery(null, {
        pollingInterval: 5000,
        refetchOnFocus: true,
        refetchOnMountOrArgChange: true
    })
```

I only see the ```console.log('working...')``` once but it should be showing up every 5 seconds because of the interval. Also, on the first call the data is loaded with correct info but then resets to null after 5 seconds.

I am trying to fetch the data using socket every x second. Thanks.

**Comment by phryneas** [comment, marked as answer]
As the name suggests, it is executed when a cache entry is added. `useGetUserInfoQuery(null)` will always share the same cache entry with every other place where you call `useGetUserInfoQuery(null)` (as both the endpoint and the argument are the same), so the cache entry is just used.
Updating an endpoint does not mean creating a new one - hence this only executes once. 

You might be looking for `onQueryStarted`, not for `onCacheEntryAdded`.
